components/style/stylesheets.rs

```diff
@@ -112,7 +120,7 @@ pub struct KeyframesRule {
 #[derive(Debug)]
 pub struct MediaRule {
     pub media_queries: Arc<RwLock<MediaList>>,
-    pub rules: Vec<CSSRule>,
+    pub rules: CSSRules,
```

Comment:
I think this would likely make it meaningless to wrap `MediaRule` in `RwLock` anymore. Probably we can do fine-grained lock everywhere...